std/experimental/allocator/building_blocks/kernighan_ritchie.d

```diff
@@ -117,13 +117,21 @@ struct KRRegion(ParentAllocator = NullAllocator)
         {
             assert(right);
             auto p = payload;
-            return p.ptr + p.length == right;
+            return p.ptr < right && right - (p.ptr + p.length) < Node.sizeof;
         }
 
-        bool coalesce()
+        bool coalesce(void* memoryEnd = null)
         {
+            // Coalesce the last node before the memory end with any possible gap
+            if (memoryEnd
+                && cast(ubyte*)memoryEnd - (payload.ptr + payload.length) < Node.sizeof)
+            {
+                size += cast(ubyte*)memoryEnd - (payload.ptr + payload.length);
```

Comment:
`assert(memoryEnd >= (payload.ptr + payload.length));` it seems

cast also seems unneeded